pageserver/src/tenant/mgr.rs

```diff
@@ -1668,19 +1668,7 @@ impl TenantManager {
         let tmp_path = safe_rename_tenant_dir(&local_tenant_directory)
             .await
             .with_context(|| format!("local tenant directory {local_tenant_directory:?} rename"))?;
-        task_mgr::spawn(
-            task_mgr::BACKGROUND_RUNTIME.handle(),
-            TaskKind::MgmtRequest,
-            None,
-            None,
-            "tenant_files_delete",
-            false,
-            async move {
-                fs::remove_dir_all(tmp_path.as_path())
-                    .await
-                    .with_context(|| format!("tenant directory {:?} deletion", tmp_path))
-            },
-        );
+        self.spawn_background_purge(tmp_path);
```

Comment:
Unrelated to this PR, but, I noticed it because of the comment on spawn_background_purge: it would be nice to 1. not rely on random numbers and 2. back-pressure ongoing background purges for a tenant.

I.e., if we `/location_conf` once that background-deletes the tenant shard directory, a second `/location_conf` for that same tenant shard should be rejected with "try again" or be coalesced on the still-ongoing background delete.

Could do that by keeping track of the background-deleting state in TenantManager.

Reply:
> I.e., if we /location_conf once that background-deletes the tenant shard directory, a second /location_conf for that same tenant shard should be rejected with "try again" or be coalesced on the still-ongoing background delete.

Hmm, I don't think this is a problem in practice in  the case of a single Tenant: for spawn_background_purge_ to be expensive there have to be a lot of layers, so if someone was e.g. calling location_config  repeatedly quickly, they wouldn't have time to download enough layers to create excessive load on local storage from the deletions.

The place we could  have issues is if someone had thousands of tenants with thousands of layers, and sent detach requests for all of them at the same time: that probably would cause some noticeable performance impact from having a very long queue of unlink calls, although we're using tokio::fs::remove_dir_all, so the concurrency would be limited by spawn_blocking pool size.

If we wanted a throttle on all this, I'd do it by using some global channel for background deletions like this -- something really simple like processing one tmp dir deletion at a time would likely work fine in practice.

As you say though, it's unrelated to this PR.